### ISSUE TITLE
Update to json newtonsoft  11.0.1

### ISSUE THREAD

**Issue by SimonCropp** [Particular/NServiceBus.Persistence.Sql#258]

**Comment by SzymonPobiega** [comment]
@SimonCropp we're looking into that but we are not yet sure how to proceed with regards to versioning.

**Comment by SimonCropp** [comment]
I think release a new stable minor and a new rc

**Comment by SimonCropp** [comment]
https://github.com/Particular/NServiceBus.Newtonsoft.Json/issues/36#issuecomment-367951397

**Comment by jukkahyv** [comment]
Hi, this is quite important for us too, because Newtonsoft.Json 11 is creeping through other nuget packages.

**Comment by SimonCropp** [comment]
bump

**Comment by SzymonPobiega** [comment]
@SimonCropp we're working on this

**Comment by EvgenyChepurnykh** [comment]
Guys, I think it's a big trouble. We use Newtonsoft.Json v11. Sql.Persistance is uncompatible but moreover NServiceBus.Newtonsoft.Json require version 11 and above. 
We have really bad situation:
NServiceBus.Persistance.Sql requires Newtonsoft.Json version less than 11.
NServiceBus.Newtonsoft.Json requires Newtonsoft.Json version above 11. 
I can't use both in one project!
How do you will repair this problem? No, that's not important.  When?

**Comment by DavidBoike** [comment]
Hi @EvgenyChepurnykh,

If you step down to version 1.1 of JSON Persistence, it uses Newtonsoft.Json 10. Then you can use that with SQL Persistence.

https://www.nuget.org/packages/NServiceBus.Newtonsoft.Json/1.1.0

As for an update to SQL Persistence, we currently have a task to release a version 3.1 of that (using JSON 11) in focus and will be shipping it soon, in case you have other dependencies pushing you toward JSON 11.

**Comment by EvgenyChepurnykh** [comment]
Yes, we have updated other projects for using Newton.Json v11 because some components require v11. Now we should do a regress to determine how our project work with different versions of json. It's not easy... Anyway, thanks for the quick response....

**Comment by EvgenyChepurnykh** [comment]
Would you say, if I use Backend with Json v.11 for sending messages and nservicebus windows services with  json v.10. Will I have some troubles with that configuration?

**Comment by DavidBoike** [comment]
AFAIK, JSON 11 has no breaking changes at the JSON (wire-compatibility) level. It's just incompatible at the binary level, hence the new major version. So I think you could confidently have one "thing" talking in JSON 10 and exchanging that output with another "thing" using JSON 11 and be just fine.

**Comment by DavidBoike** [comment]
P.S. But of course I would test that. :wink:

**Comment by EvgenyChepurnykh** [comment]
Thanks for the answers and yes we will do a regress in any cases.

**Comment by EvgenyChepurnykh** [comment]
But we will wait for the update on your side :)

**Comment by SimonCropp** [comment]
IMO blocking people from updating such a central dependency for over a month is not satifactory.

**Comment by SzymonPobiega** [comment]
We're releasing 3.1.0 and 4.1.0-beta1

**Comment by SzymonPobiega** [comment]
3.1.0 and 4.1.0-beta1 have been released.

**Comment by EvgenyChepurnykh** [comment]
Thanks :)

**Comment by SimonCropp** [comment]
@SzymonPobiega thanks heaps
